tests/test_game.py

```diff
@@ -294,3 +294,660 @@ def test_tie_board_is_game_over(self):
+
+    def test_move_count_default(self):
+        assert self.C4._move_count == 0
+
+    def test_board_state_default(self):
+        assert torch.all(self.C4.board_state.eq(-1))
+
+    def test_dims_custom(self):
+        C4 = Connect4(width=8, height=7)
+
+        assert C4.board_state.shape == (7, 8)
+        assert C4.width == 8
+        assert C4.height == 7
+
+    def test_board_state_custom(self):
+        board = self.dummy_tensor
+        C4 = Connect4(board_state=board)
+
+        assert torch.all(C4.board_state.eq(board))
+
+    def test_make_move(self):
+        self.C4.make_move(0)
+
+        result = self.dummy_tensor
+
+        assert torch.all(self.C4.board_state.eq(result))
+
+        with pytest.raises(TypeError):
+            self.C4.make_move([3, 3])
+
+        with pytest.raises(TypeError):
+            self.C4.make_move((1, 1))
+
+        with pytest.raises(IndexError):
+            self.C4.make_move(10)
+
+        self.C4.reset()
+
+    def test_board_after_move(self):
+        initial = Connect4.get_initial_board()
+        board = Connect4.board_after_move(initial, 0, 0)
+
+        assert torch.all(board.eq(self.dummy_tensor))
+
+        board = Connect4.board_after_move(board, 1, 1)
+
+        next_state = torch.Tensor(
+            [
+                [-1, -1, -1, -1, -1, -1, -1],
+                [-1, -1, -1, -1, -1, -1, -1],
+                [-1, -1, -1, -1, -1, -1, -1],
+                [-1, -1, -1, -1, -1, -1, -1],
+                [-1, -1, -1, -1, -1, -1, -1],
+                [0, 1, -1, -1, -1, -1, -1],
+            ]
+        )
+
+        final_state = torch.Tensor(
+            [
+                [0, -1, -1, -1, -1, -1, -1],
+                [1, -1, -1, -1, -1, -1, -1],
+                [0, -1, -1, -1, -1, -1, -1],
+                [1, -1, -1, -1, -1, -1, -1],
+                [-0, -1, -1, -1, -1, -1, -1],
+                [1, -1, -1, -1, -1, -1, -1],
+            ]
+        )
+
+        assert torch.all(board.eq(next_state))
+
+        with pytest.raises(TypeError):
+            Connect4.board_after_move(initial, 0, (1, 1))
+
+        with pytest.raises(TypeError):
+            Connect4.board_after_move(initial, 0, [1, 1])
+
+        with pytest.raises(ValueError):
+            Connect4.board_after_move(final_state, 0, 0)
+
+        with pytest.raises(ValueError):
+            Connect4.board_after_move(initial, 2, 1)
+
+        with pytest.raises(IndexError):
+            Connect4.board_after_move(initial, 0, 10)
+
+    def test_get_initial_board(self):
+        board = Connect4.get_initial_board()
+        assert torch.all(board.eq(-1))
+        assert board.shape == (6, 7)
+
+    def test_1(self):
+        legal_moves = Connect4.get_legal_moves(self.dummy_tensor)
+
+        assert torch.all(legal_moves.eq(self.dummy_tensor[0, :] == -1))
+
+        legal_moves = Connect4.get_legal_moves(Connect4.get_initial_board())
+
+        assert torch.all(legal_moves.eq(True))
```

Comment:
not sure what this test is doing, could you explain?

Reply:
To be honest, I took this test(the name) from your TTT tests. Just makes sure get_legal_moves, I'll rename it for TTT as well.

Reply:
LOL i'm dumb, i guess.